objectiveC-day7.txt

```diff
@@ -0,0 +1,195 @@
+ 
+b) manual reference counting:
+
+     If you do not want to depend on ARC (automatic reference counting),then you need to know how to deal with reference counts.
+     [object retain]  To increment  a reference count by 1.
+     [object release]  to decrement the reference count by 1.
+     When the reference count reaches 0,the system knows the object is no longer being used(not referenced anywhere),so it frees up memory
+     by sending the object  a dealloc message.
+
+
+
+Memory management using automatic reference counting:
+Arc was introduced in ios version 4.
+System determines when to retain an object,release or when to autorelease.Under the
+hood, reference counts are still being maintained and tracked.
+Programmers do not need to manually perform 'retain' and 'release' operations.
+
+-------------------------------------------------------------------------------------------------------------------------------------------------
+
+
+2)
+Strong Variables,properties:
+
+---By default, all object pointer variables are strong variables. That means that assigning an object
+reference to such a variable causes that object to be automatically retained and the old
+object reference will be released before the assignment is made.
+---Strong variables are initialized to zero by default.
+
+Illustration:
+Student *c1 = [[Student alloc] init];
+Student *c2 = [[Student alloc] init];
+ [c1 setName:@"bb"];
+ [c2 setName:@"bb"];
+
+Now consider the below assignment using manual memory management.
+c2=c1
+
+the effect is to just copy the reference of object f1 into f2. The Student object
+referenced by c2 would be lost as a result, as its value is overwritten. This creates a
+ memory leak  (a variable that is no longer referenced and therefore can’t be released).
+If you’re using ARC, c1 and c2 are both strong variables. So, the previous assignment actually
+works like this:
+ [c1 retain]; // retain new value
+ [c2 release]; // release the old value
+ c2 = c1; // copy the reference
+ --The above steps are done by the compiler automatically.
+-Properties that we write inside a class are not Strong by default.We have to declare them explicitly.
+example: @property(strong) NSMutableArray *employeList
+
+Weak variables:
+-Weak variables come into usage when there is a retain cycle.Retain cycle is created when two objects have reference to each other.
+When a parent class holds a reference to a subclass and subclass holds a reference to parent class,a cycle is created and these objects can
+never be destroyed.So,we need to make another type of reference known as weak reference between two objects.
+--A weak variable does not prevent deallocation of the object it references.
+Syntac: @property (weak) NSMutableArray *employeList;
+
+--weak variables are not supported in iOS 4 or Mac OS X 10.6. In such cases, you can
+still use the unsafe_unretained (or assign) property attribute or declare your variable to be __unsafe_unretained.
+
```

Comment:
I guess you have misunderstood about string qualifier.

Declaring a property/ variable as string signifies that it want to take ownership of the object. It increase the reference count of object by 1.

Re-read about strong and weak properties and update your answer.
Also explain how declaring property/variable as weak affect the reference count of the object.